Add Whatsapp loader

batmanscode opened the pull request:
I was already working with Whatsapp data and seeing the competition message on https://llamahub.ai/ motivated me to add this in quick! 😄

batmanscode commented:
Will close #28

emptycrown commented on loader_hub/whatsapp/base.py:
```diff
@@ -0,0 +1,57 @@
+from gpt_index.readers.schema.base import Document
+
+
+class WhatsappChatLoader(BaseReader):
```
You'll need to add this to the `library.json` file as well in order for it to be picked up by `download_loader` in LlamaIndex and the LlamaHub website.

batmanscode replied:
Missed this! Thanks

emptycrown commented on loader_hub/whatsapp/base.py:
```diff
@@ -0,0 +1,57 @@
+        parser.parse_file()
+        df = parser.parsed_messages.get_df()
+
+        print(f"Number of messages: {len(df)}.")
```
Can we use `logging` here instead of `print`? You can search the codebase for a few examples.

batmanscode replied:
Can do. Lol as I was submitting the PR I was wondering about this

emptycrown commented on loader_hub/whatsapp/base.py:
```diff
@@ -0,0 +1,57 @@
+
+            if verbose:
+                n += 1
+                print(f"Added {n} of {len(df)} messages.")
```
Same as above. `import logging` etc is better.

emptycrown commented on loader_hub/whatsapp/base.py:
```diff
@@ -0,0 +1,57 @@
+        
+        self.file_path = path
+
+    def load_data(self, verbose=False) -> List[Document]:
```
Using `logging` also removes the need to have a `verbose` argument here, so we can take that out.

emptycrown reviewed (changes requested):
Thanks a bunch for doing this! Nice to see you knock out one of the open requests. 

Left a few comments.

batmanscode commented:
> Thanks a bunch for doing this! Nice to see you knock out one of the open requests.
> 
> Left a few comments.

Great library, glad to be able to contribute. And thanks! 😃

Addressed your comments 👍, let me know if there's anything I missed

batmanscode commented:
Thanks for the quick feedback and merge @emptycrown. I think this is the fasted my PR has ever been merged! 😆
